packages/builder/src/components/automation/SetupPanel/FieldSelector.svelte

```diff
@@ -38,77 +47,58 @@
     NUMBER: null,
     DATETIME: null,
     STRING: "",
-    OPTIONS: [],
-    ARRAY: [],
+    ARRAY: "",
   }
 
-  const coerce = (value, type) => {
-    const re = new RegExp(/{{([^{].*?)}}/g)
-    if (re.test(value)) {
-      return value
-    }
-
-    if (type === "boolean") {
-      if (typeof value === "boolean") {
-        return value
-      }
-      return value === "true"
-    }
-    if (type === "number") {
-      if (typeof value === "number") {
-        return value
-      }
-      return Number(value)
+  const onChange = (e, field) => {
+    if (e.detail !== editableValue[field]) {
+      editableValue[field] = e.detail
+      dispatch("change", editableValue)
     }
-    if (type === "options") {
-      return [value]
-    }
-    if (type === "array") {
-      if (Array.isArray(value)) {
-        return value
-      }
-      return value.split(",").map(x => x.trim())
-    }
-
-    if (type === "link") {
-      if (Array.isArray(value)) {
-        return value
-      }
-
-      return [value]
-    }
-
-    return value
-  }
-
-  const onChange = (e, field, type) => {
-    value[field] = coerce(e.detail, type)
-    dispatch("change", value)
   }
 </script>
 
-{#if schemaFields.length && isTestModal}
-  <div class="schema-fields">
+{#if schemaFields?.length && isTestModal}
```

Comment:
Is this component ever actually used? Was having trouble triggering it to render, only place it's consumed is 
[here](https://github.com/Budibase/budibase/blob/feature/automation-row-ux-update/packages/builder/src/components/automation/SetupPanel/AutomationBlockSetup.svelte#L950) seemingly, but I couldn't get it to render with any of the automation step types I tried.

Reply:
This component is found in `App action` trigger types. It's only visible in the test modal. It renders the fields in the trigger and makes them editable.